### PR TITLE
Navigation Rail preview

### PR DESCRIPTION
**Pull request by imashnake0**
[comment]: # (Replace [ ] with [x].)
- [x] Read [`CONTRIBUTING.md`](https://github.com/imashnake0/Animite/blob/be53ba4561c8ff20f588fb348965f208e301b6b8/CONTRIBUTING.md).

Preview of Navigation Rail.

**Summary of changes:**

1. Created `NavigationRail`.

**Tested changes:**

♥\\(灬♡ ◡ ♡灬\\)

[comment]: # (THANK YOU! ♡\(灬♥ ◡ ♥灬\))

### CODE REVIEW

**Comment by imashnake0** [comment]
Screenshot_20230215-122930

File: app/src/main/java/com/imashnake/animite/features/MainActivity.kt [hunk 1]
```diff
@@ -127,13 +136,24 @@ fun MainScreen(modifier: Modifier = Modifier) {
...
         )
 
-        AnimatedVisibility(
-            visible = isNavBarVisible,
-            modifier = Modifier.align(Alignment.BottomCenter),
-            enter = slideInVertically { it },
-            exit = slideOutVertically { it }
-        ) {
-            NavigationBar(navController = navController)
+        if (LocalConfiguration.current.orientation == Configuration.ORIENTATION_PORTRAIT) {
```
Comment (boswelja) [inline app/src/main/java/com/imashnake/animite/features/MainActivity.kt:139]: This block is a good candidate to be extracted to a separate Composable

File: app/src/main/java/com/imashnake/animite/features/MainActivity.kt [hunk 2]
```diff
@@ -127,13 +136,24 @@ fun MainScreen(modifier: Modifier = Modifier) {
...
+            AnimatedVisibility(
+                visible = isNavBarVisible,
+                modifier = Modifier.align(Alignment.BottomCenter),
+                enter = slideInVertically { it },
+                exit = slideOutVertically { it }
+            ) {
+                NavigationBar(navController = navController)
```
Comment (boswelja) [inline app/src/main/java/com/imashnake/animite/features/MainActivity.kt:146]: Can we just have the if/else inside AnimatedVisibility?

File: app/src/main/res/values/dimens.xml [hunk 3]
```diff
@@ -9,6 +9,7 @@
     <dimen name="media_card_text_padding_vertical">10dp</dimen>
 
     <dimen name="navigation_bar_height">65dp</dimen>
+    <dimen name="navigation_rail_width">80dp</dimen>
```
Comment (boswelja) [inline app/src/main/res/values/dimens.xml:12]: Why do we define sizes for these anyways

File: app/src/main/java/com/imashnake/animite/features/MainActivity.kt (hunk 2 above)
Comment (imashnake0) [reply app/src/main/java/com/imashnake/animite/features/MainActivity.kt:146]: The enter and exit animations are different, can't tell which one is better and why.

File: app/src/main/res/values/dimens.xml (hunk 3 above)
Comment (imashnake0) [reply app/src/main/res/values/dimens.xml:12]: Because we're not using a Scaffold :kappa:

Comment (boswelja) [reply app/src/main/res/values/dimens.xml:12]: Scaffold doesn't constrain the size either, size is set by the component

Comment (imashnake0) [reply app/src/main/res/values/dimens.xml:12]: How would I add Navigation Bar padding to the other components?

Comment (boswelja) [reply app/src/main/res/values/dimens.xml:12]: Idk, stop using a box

Comment (imashnake0) [reply app/src/main/res/values/dimens.xml:12]: o7

**Comment by imashnake0** [comment]
Going to work on a custom layout to accommodate this.

**Pull request closed without merging**